lib/OpenQA/Command.pm

```diff
@@ -89,6 +90,13 @@ sub parse_params {
         push @{$params{$1}}, $2;
     }
 
+    return \%params unless (defined $params{template});
+
+    foreach my $tval (@{$params{template}}) {
```

Comment:
I feel like `reference` should get the same treatment since you will run into the same problems.

Reply:
If you haven't used it: `reference` is the expected current YAML document that the openQA instance knows about to avoid race conditions.

Reply:
But isn't this `reference` in the openQA instance itself? I mean, on the server side? Not sure how this is impacted by command line arguments processing on the client side.

Reply:
I changed that part of the code. Not sure if your comment still applies.

Reply:
Yeah, now it is not needed anymore. But with the previous approach it would have been, since the `reference` parameter can be given in the same way as the `template` parameter to avoid overwriting intermediate changes.

Reply:
Ah, now I get it. `reference` is also a parameter.